pghoard/transfer.py

```diff
@@ -147,14 +147,32 @@ def untrack_upload_event(self, file_key: str) -> None:
 
     def increment(self, file_key: str, total_bytes_uploaded: float) -> None:
         metric_data = {}
+        persisted_progress = PersistedProgress.read(metrics=self.metrics)
+
         with self._tracked_events_lock:
             if file_key not in self._tracked_events:
                 raise Exception(f"UploadEvent for {file_key} is not being tracked.")
 
             file_type = self._tracked_events[file_key].file_type
             if file_type in (
-                FileType.Basebackup, FileType.Basebackup_chunk, FileType.Basebackup_delta, FileType.Basebackup_delta_chunk
+                FileType.Basebackup,
+                FileType.Basebackup_chunk,
+                FileType.Basebackup_delta,
+                FileType.Basebackup_delta_chunk,
             ):
+                progress_info = persisted_progress.get(file_key)
+                if total_bytes_uploaded > progress_info.current_progress:
+                    progress_info.update(total_bytes_uploaded)
+                    persisted_progress.write(metrics=self.metrics)
+                    self.metrics.gauge("pghoard.basebackup_stalled", 0)
+                else:
+                    stalled_age = progress_info.age
+                    self.metrics.gauge("basebackup_stalled", stalled_age)
+                    self.log.warning(
```

Comment:
This shouldn't be a warning. Or at least it shouldn't be warning until we meet some threshold

Reply:
use log.info or use a threshold for  5 min

Reply:
added a 10 min threshold